### ISSUE TITLE
rustup-init skip installing additional target if default toolchain exists

### ISSUE THREAD

**Issue by tesuji** [rust-lang/rustup#2215]
**Problem**

`rustup-init` skips adding additional toolchain via `--target` option when default-toolchain exists.

*Currently*: rustup-init updates itself and stop
*Expected*: rustup should add the `target` if it doesn't exist.
*Context*: I run a script on CI that makes use of `rustup-init` despite of rustup exist or not.

**Steps**

1. Install stable toolchain
```console
rustup install stable
```
2. Try using `rustup-init`
```console
$ rustup-init.exe --default-host=x86_64-pc-windows-msvc --default-toolchain=stable --target=i686-pc-windows-msvc
info: profile set to 'minimal'
info: setting default host triple to x86_64-pc-windows-msvc
info: updating existing rustup installation


Rust is installed now. Great!

To get started you need Cargo's bin directory (%USERPROFILE%\.cargo\bin) in your
PATH
environment variable. Future applications will automatically have the
correct environment, but you may need to restart your current shell.
$ cargo build --target i686-pc-windows-msvc
    Updating crates.io index
   Compiling winapi v0.3.8
error[E0463]: can't find crate for `core`
  |
  = note: the `i686-pc-windows-msvc` target may not be installed

error: aborting due to previous error

For more information about this error, try `rustc --explain E0463`.
error: could not compile `winapi`.

To learn more, run the command again with --verbose.
```

**Possible Solution(s)**

**Notes**

Output of `rustup --version`: rustup 1.21.1 (7832b2ebe 2019-12-20)
Output of `rustup show`:

**Comment by kinnison** [comment]
This is already being addressed in #2201 which you have commented on, but has not yet been merged.

**Comment by tesuji** [comment]
Thank you. I totally forgot about that PR.
